.github/workflows/validate-all.yml

```diff
@@ -16,53 +16,92 @@ on:
+          pip install wheel
+          pip install git+https://github.com/bigbio/sdrf-pipelines
+      - name: Get a list of changed files
+        if: github.event_name != 'schedule'
+        id: getfiles
+        run: |
+          export OUTFILE=$HOME/filelist.txt
+          if [ "${{ github.event_name }}" == pull_request ]; then
+              echo 'Changed files:'
+              git diff-tree --no-commit-id --name-only -r origin/${{ github.event.pull_request.base.ref }} ${{ github.sha }} | tee "$OUTFILE"
           else
-            parent="${{ github.event.before }}"
+            # push
+            if [ "${{ github.event.before }}" == "0000000000000000000000000000000000000000" ]; then
+              parent="origin/master" # compare with master as a last resort
+            else
+              parent="${{ github.event.before }}"
+            fi
+            git diff-tree --no-commit-id --name-only -r "$parent" ${{ github.sha }} | tee "$OUTFILE"
           fi
-          git diff-tree --no-commit-id --name-only -r "$parent" ${{ github.sha }} | tee "$OUTFILE"
-        fi
-    - name: Run validation
-      run: |
-        if [ ${{ github.event_name }} == "schedule" ]; then
-          echo "Validating all projects (schedule)"
-          python validate.py -v
-        else
-          export FILELIST=$HOME/filelist.txt
-          cat "$FILELIST"
-          if grep -q '^validate.py' "$FILELIST"; then
-            echo "Validating all projects (script changed)"
+      - name: Run validation
+        run: |
+          if [ ${{ github.event_name }} == "schedule" ]; then
+            echo "Validating all projects (schedule)"
             python validate.py -v
           else
-            projects=$(cat "$FILELIST" | grep '^annotated-projects' | cut -d/ -f 2)
-            echo "Validating projects:" $projects
-            python validate.py -v $projects
+            export FILELIST=$HOME/filelist.txt
+            cat "$FILELIST"
+            if grep -q '^validate.py' "$FILELIST"; then
+              echo "Validating all projects (script changed)"
+              python validate.py -v
+            else
+              projects=$(cat "$FILELIST" | grep '^annotated-projects' | cut -d/ -f 2)
+              echo "Validating projects:" $projects
+              python validate.py -v $projects
+            fi
           fi
-        fi
+  mage_tab_validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup conda
+        uses: s-weigand/setup-conda@v1
+        with:
+          update-conda: true
+          conda-channels: anaconda, bioconda, conda-forge
+      - name: Install perl-atlas-modules
+        run: conda create -n perl-atlas -c ebi-gene-expression-group -c bioconda perl-atlas-modules perl-graphviz
+      - name: Validate General MAGE-TAB
+        run: |
+          . /usr/share/miniconda/bin/activate /usr/share/miniconda/envs/perl-atlas
+          perl simple_validate_magetab.pl
```

Comment:
I suspect that this should be receiving an argument, or at least the code does. Does this work well with no arguments?